cmd/entrypoint/secrets.go

```diff
@@ -313,6 +323,59 @@ func ReconcileSecrets(ctx context.Context, sh *SnapshotHolder) error {
 	return nil
 }
 
+// Returns secretName, secretNamespace from a provided (unstructured) filter if it contains a secret
+// Returns empty strings when the secret name and/or namespace could not be found
+func findFilterSecret(filter *unstructured.Unstructured, action func(snapshotTypes.SecretRef)) error {
+	// Just making extra sure this is actually a Filter
+	if filter.GetKind() != "Filter" && filter.GetKind() != "Filters" {
+		return errors.New("Non-Filter object in Snapshot.Filters")
+	}
+	// Only Oauth2 Filters have secrets, although they dont need to have them.
+	// This is overly contrived because Filters are unstructured to emissary since we dont have the type definitions
+	// Yes this is disgusting. It is what it is...
+	filterContents := filter.UnstructuredContent()
+	filterSpec := filterContents["spec"]
+	if filterSpec != nil {
+		mapOauth, ok := filterSpec.(map[string]interface{})
+		// We need to check if all these type assertions fail since we shouldnt rely on CRD validation to protect us from a panic state
+		// I cant imagine a scenario where this would realisticly happen, but we generate a unique log message for tracability and skip processing it
+		if !ok {
+			return errors.New("Filter object detected with a bogus \"spec\" field")
+		}
+		oauthFilter := mapOauth["OAuth2"]
+		if oauthFilter != nil {
+			secretName, secretNamespace := "", ""
+			// Check if we have a secretName
+			mapOauth, ok := oauthFilter.(map[string]interface{})
+			if !ok {
+				return errors.New("Filter object detected with a bogus \"OAuth2\" field")
+			}
+			oauthSecretName := mapOauth["secretName"]
+			// This is a weird check, but we have to handle the case where secretName is not provided, and when its explicitly set to ""
+			if sName, ok := oauthSecretName.(string); ok && sName != "" {
+				secretName = sName
+			} else {
+				if !ok && oauthSecretName != nil {
+					return errors.New("Filter object detected with a bogus \"secretName\" field")
+				}
```

Comment:
IMO having a specific error here is making the code hard to read for little benefit. The apiserver should catch+report this. apiext should catch+report this. amb-sidecar should catch+report this. No need to report it again here.